gpMgmt/bin/gprebalance_modules/rebalance_commons.py

```diff
@@ -1,121 +1,56 @@
-            return str(cursor.fetchone()[0])
-        return 'not defined'
-
-    def rebalanceSchema(self, target_segment_count: int) -> None:
-        # Before rebalancing check if the tables are already rebalanced
-        # (in case we re-enter after interruption that happened after COMMIT but before new state)
-        if get_table_distr_segment_count(self.conn, self.schema_name, self.rebalance_status) > target_segment_count:
-            dbconn.execSQL(self.conn,
-                           f'''ALTER TABLE "{self.schema_name}"."{self.rebalance_status}"
-                           REBALANCE {target_segment_count}''')
-
-        if get_table_distr_segment_count(self.conn, self.schema_name, self.table_rebalance_status_detail) > target_segment_count:
-            dbconn.execSQL(self.conn,
-                           f'''ALTER TABLE "{self.schema_name}"."{self.table_rebalance_status_detail}"
-                           REBALANCE {target_segment_count}''')
-
-        if get_table_distr_segment_count(self.conn, self.schema_name, self.saved_plan) > target_segment_count:
-            dbconn.execSQL(self.conn,
-                           f'''ALTER TABLE "{self.schema_name}"."{self.saved_plan}"
-                           REBALANCE {target_segment_count}''')
-
-    def storeState(self, state: str) -> None:
-        if self.schemaExists():
-            dbconn.execSQL(self.conn,
-                           f'''INSERT INTO {self.schema_name}.{self.rebalance_status}
-                           VALUES ('{state}', NOW())''')
-
-    def clearTablesToRebalanceWithStatus(self, status: str) -> None:
-        dbconn.execSQL(self.conn,
-                       f'''DELETE FROM {self.schema_name}.{self.table_rebalance_status_detail}
-                       WHERE (status = '{status}')''')
-
-    def addTableToRebalance(self, db: str, schema_name: str, rel_name: str, status: str) -> None:
-        dbconn.execSQL(self.conn,
-                       f'''INSERT INTO {self.schema_name}.{self.table_rebalance_status_detail}
-                       VALUES ('{db}', '{schema_name}', '{rel_name}', '{status}')''')
-
-    def setStatusForTableToRebalance(self, db: str, schema_name: str, rel_name: str, status: str) -> None:
-        dbconn.execSQL(self.conn,
-                       f'''UPDATE {self.schema_name}.{self.table_rebalance_status_detail} SET status = '{status}'
-                       WHERE db_name = '{db}' AND schema_name = '{schema_name}' AND rel_name = '{rel_name}';''')
-
-    def getTablesToRebalanceWithStatus(self, status: str) -> cursor:
-        return dbconn.query(self.conn, f"""SELECT db_name, schema_name, rel_name FROM
-                            {self.schema_name}.{self.table_rebalance_status_detail} WHERE status = '{status}'""")
+from dataclasses import dataclass
+from typing import Set, Optional
+from enum import IntEnum
+from gppylib.gparray import Segment 
+
+class HostStatus(IntEnum):
+    ACTIVE = 1
+    NEW = 2
+    DECOMMISSION = 3

```

Comment:
If speaking about status of some host, better
```suggestion
    DECOMMISSIONED = 3
```